src/components/AddNewFood.js

```diff
@@ -0,0 +1,91 @@
+import React, { Component } from 'react';
+
+export default class AddNewFood extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      name: '',
+      calories: '',
+      image: '',
+      quantity: '',
+    };
+  }
+
+  handleNameInput = e => {
+    this.setState({
+      name: e.target.value,
+    });
+  };
+
+  handleCaloriesInput = e => {
+    this.setState({
+      calories: e.target.value,
+    });
+  };
+
+  handleImageInput = e => {
+    this.setState({
+      image: e.target.value,
+    });
+  };
+
+  handleQuantityInput = e => {
+    this.setState({
+      quantity: e.target.value,
+    });
+  };
+
+  handleFormSubmit = e => {
+    e.preventDefault();
+    this.props.addNew(this.state);
+    this.setState({
+      name: '',
+      calories: '',
+      quantity: '',
+      image: '',
+    });
+  };
+
+  render() {
+    return (
+      <div>
+        <form onSubmit={this.handleFormSubmit}>
+          <label>Food Name:</label>
+          <input
+            type="text"
+            name="name"
+            value={this.state.name}
+            onChange={e => this.handleNameInput(e)}
```

Comment:
No necesitas hacer esto `onChange={e => this.handleNameInput(e)}`, cuando llamas a una funcion a través de un evento, la función lo recibe intrinsecamente como parámetro: `onChange={this.handleNameInput}`